Simplenote/src/androidTest/java/com/automattic/simplenote/integration/TagDialogFragmentTest.kt

```diff
@@ -0,0 +1,326 @@
+
+        onView(allOf(withText("tag2"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText("tag10"))
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(isDisplayed())).perform(click())
+
+        val tag1 = getTag("tag1")
+        val tag2 = getTag("tag2")
+        val tag3 = getTag("tag3")
+        val tag10 = getTag("tag10")
+
+        assertNull(tag2)
+        assertNotNull(tag1)
+        assertNotNull(tag3)
+        assertNotNull(tag10)
+
+        assertEquals(note1.tags, listOf("tag1", "tag10"))
+        assertEquals(note2.tags, listOf("tag3", "tag10"))
+        assertEquals(note3.tags, listOf("tag1", "tag3"))
+    }
+
+    @Test
+    fun editTryMergeTagInMultipleNotes() {
+        createTag("tag1")
+        createTag("tag2")
+        createTag("tag3")
+        // To edit tags, tags should belong a note
+        val note1 = createNote("Hello1", listOf("tag1", "tag2"))
+        val note2 = createNote("Hello1", listOf("tag2", "tag3"))
+        val note3 = createNote("Hello1", listOf("tag1", "tag3"))
+
+        assertEquals(tagsBucket.count(), 3)
+        assertEquals(notesBucket.count(), 3)
+
+        launchFragment("tag2")
+
+        onView(allOf(withText("tag2"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        onView(allOf(withText("tag2"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText("tag3"))
+
+        onView(allOf(withText("tag3"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(isDisplayed())).perform(click())
+
+        val canonical = TagUtils.getCanonicalFromLexical(tagsBucket, "tag3")
+        val mergeMessage = getResourceStringWithArgs(R.string.dialog_tag_conflict_message, canonical, "tag2", canonical)
+        onView(withId(R.id.message)).check(matches(withText(mergeMessage)))
+
+        val backText = getResourceString(R.string.back)
+        onView(withText(backText)).perform(click())
+
+        val cancelText = getResourceString(R.string.cancel)
+        onView(withText(cancelText)).inRoot(isDialog()).perform(click())
+
+        val tag1 = getTag("tag1")
+        val tag2 = getTag("tag2")
+        val tag3 = getTag("tag3")
+
+        assertNotNull(tag2)
+        assertNotNull(tag1)
+        assertNotNull(tag3)
+
+        assertEquals(note1.tags, listOf("tag1", "tag2"))
+        assertEquals(note2.tags, listOf("tag2", "tag3"))
+        assertEquals(note3.tags, listOf("tag1", "tag3"))
+    }
+
+    @Test
+    fun editMergeTagInMultipleNotes() {
```

Comment:
Question (❓): This test is missing from the test cases within [1364](https://github.com/Automattic/simplenote-android/issues/1364) issue. Isn't it?

Reply:
Yes, that should be test 7. I have added the test to the issue.